### PR TITLE
Fix Windows llvmdev build due to zlib linkage

### PR DESCRIPTION
**Pull request by sklam**
(Based on #1104)

Users have reported error using the windows wheel for llvmlite on official CPython distribution (non-Anaconda builds). The error was due to `zlib.dll` dependency that is not available on the installed Python. 

This patch fixes the problem by using `llvmdev_manylinux` recipe which disabled zlib and zstd.

Note: With this patch, all wheel builds use `llvmdev_manylinux` recipe. The recipe folder should rename to `llvmdev_wheels` later. Don't want to do that now which will require changing more buildfarm scripts.

In addition, a GHA workflow is added to build the llvmdev for wheels. Example run: https://github.com/sklam/llvmlite/actions/runs/12186143333/job/33993961959

### CODE REVIEW

File: .github/workflows/llvmdev_win_builder.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: llvmdev_win_builder
```
Comment (sklam) [inline .github/workflows/llvmdev_win_builder.yml:1]: This is a adaptation of https://github.com/numba/llvmlite/pull/1106/files#diff-1db27d93186e46d3b441ece35801b244db8ee144ff1405ca27a163bfe878957fR162-R207

This workflow is saved here for record keeping purpose. No automatic trigger.

This workflow was used to produce the llvmdev_manylinux build on https://anaconda.org/numba/llvmdev/15.0.7/download/win-64/llvmdev-15.0.7-manylinux.tar.bz2

File: conda-recipes/llvmdev_manylinux/bld.bat [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+cmake -G "Ninja" ^
+    -DCMAKE_BUILD_TYPE="Release" ^
+    -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX% ^
+    -DCMAKE_INSTALL_PREFIX:PATH=%LIBRARY_PREFIX% ^
+    -DLLVM_USE_INTEL_JITEVENTS=ON ^
+    -DLLVM_ENABLE_LIBXML2=OFF ^
+    -DLLVM_ENABLE_RTTI=ON ^
+    -DLLVM_ENABLE_ZLIB=OFF ^
+    -DLLVM_ENABLE_ZSTD=OFF ^
```
Comment (sklam) [inline conda-recipes/llvmdev_manylinux/bld.bat:20]: This file is a copy of `llvmdev/bld.bat` with these two lines changed.

**Review by swap357 — COMMENTED** [review]
I don't fully understand the implications of recipe changes but workflow looks good to me.

File: .github/workflows/llvmdev_win_builder.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+jobs:
+
+  windows:
+    name: Windows • Py${{ matrix.config.python }}
+    runs-on: windows-2019
+    strategy:
+      matrix:
+        config:
+          - {python: '3.10'}
```
[the remark this answers is not in the document]
Comment (sklam) [reply .github/workflows/llvmdev_win_builder.yml]: Simplifed. New GHA test run: https://github.com/sklam/llvmlite/actions/runs/12258663138/job/34198957127

**Review by kc611 — APPROVED** [review]
LGTM

**Review by kc611 — DISMISSED** [review]

**Review by kc611 — APPROVED** [review]

**Review by kc611 — DISMISSED** [review]

**Review by kc611 — APPROVED** [review]

**Pull request merged**